apps/recovery-utility/renderer/lib/wallets/TON/index.ts

```diff
@@ -0,0 +1,52 @@
+import { Ton as BaseTon, Input } from '@fireblocks/wallet-derivation';
+import { SigningWallet } from '../SigningWallet';
+import { Address, beginCell, Cell, toNano } from '@ton/core';
+import { GenerateTxInput, TxPayload } from '../types';
+
+export class Ton extends BaseTon implements SigningWallet {
+  constructor(input: Input) {
+    super(input);
+  }
+
+  public async generateTx({ to, amount, feeRate, memo, extraParams }: GenerateTxInput): Promise<TxPayload> {
+    // calculate the amount to withdraw
+    const fee = BigInt(toNano(feeRate!)); // feeRate as BigInt in nano
+    const amountToWithdraw = BigInt(toNano(amount)) - fee; // amount is the wallet balance
+    let internalMessageMemo = undefined;
+    if (memo) {
+      internalMessageMemo = beginCell().storeUint(0, 32).storeStringTail(memo).endCell();
+    }
+    // create the tx payload
+    let internalMessage = beginCell()
+      .storeUint(0x10, 6) // 0x10 is no bounce
+      .storeAddress(Address.parse(to)) // Store the recipient address
+      .storeCoins(amountToWithdraw); // Store the amount within the payload
+
+    if (internalMessageMemo) {
+      internalMessage
+        .storeUint(1, 1 + 4 + 4 + 64 + 32 + 1 + 1) // store memo as reference
+        .storeRef(internalMessageMemo)
+        .endCell();
+    } else {
+      internalMessage.storeUint(0, 1 + 4 + 4 + 64 + 32 + 1 + 1).endCell(); // no memo added
+    }
+
+    let toSign = beginCell()
+      .storeUint(698983191, 32) // sub wallet_id
```

Comment:
Is there a reference for this? if so, please put it in a comment so we can check in case it changes. in the future